airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/upgrade_java_cdk/pipeline.py

```diff
@@ -0,0 +1,99 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+from __future__ import annotations
+
+import os
+import re
+from typing import TYPE_CHECKING
+
+from pipelines.airbyte_ci.connectors.context import ConnectorContext
+from pipelines.airbyte_ci.connectors.reports import ConnectorReport, Report
+from pipelines.helpers import git
+from pipelines.models.steps import Step, StepResult, StepStatus
+
+if TYPE_CHECKING:
+    from anyio import Semaphore
+
+
+class SetJavaCDKVersion(Step):
+    context: ConnectorContext
+    title = "Set Java CDK Version"
+
+    def __init__(
+        self,
+        context: ConnectorContext,
+        new_version: str,
+    ) -> None:
+        super().__init__(context)
+        self.new_version = new_version
+
+    async def _run(self) -> StepResult:
+        context = self.context
+
+        target_java_cdk_version = target_java_cdk_version
+        if target_java_cdk_version == "":
+            cdk_version_properties = context.get_repo_file("/airbyte-cdk/java/airbyte-cdk/core/src/main/resources/version.properties")
+            version_properties_content = cdk_version_properties.contents()
+            target_java_cdk_version = re.search(r"version *= *(?P<version>[0-9]*\.[0-9]*\.[0-9]*)", og_build_gradle_content).group(
+                "version"
+            )
```

Comment:
Does this mean: if the target java cdk version is not set we'll use the latest one declared in the CDK?
Can we make this logic more obvious by isolating this inside its own method? Can we also make `target_java_cdk_version` `None` instead of an empty string?